kong/db/schema/init.lua

```diff
@@ -1614,8 +1614,18 @@ function Schema:process_auto_fields(data, context, nulls)
     end
 
     for key in pairs(data) do
-      -- set non defined fields to nil, except meta fields (ttl) if enabled
-      if not self.fields[key] then
+      local field = self.fields[key]
+      if field then
+        if not field.legacy
```

Comment:
Legacy fields are UUID fields, right? Why are legacy fields `null` by default and not `nil`? See https://github.com/Kong/kong/blob/master/kong/db/schema/init.lua#L1561-L1563